### PR TITLE
[Merged by Bors] - feat: ind-objects are closed under products

### PR DESCRIPTION
**Pull request by TwoFX**
---

- [x] depends on: #18979
- [x] depends on: #18980
- [x] depends on: #18981
- [x] depends on: #18982
- [x]  depends on: #18987

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Limits/FilteredColimitCommutesProduct.lean [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+import Mathlib.CategoryTheory.Limits.FunctorCategory.Shapes.Products
+
+/-!
+# The IPC property
+
+Given a family of categories `I i` (`i : α`) and a family of functors `F i : I i ⥤ C`, we construct
+the natural morphism `colim_k (∏ᶜ s ↦ (F s).obj (k s)) ⟶ ∏ᶜ s ↦ colim_k (F s).obj (k s)`.
+
+Similarly to the study of finite limits commuting with filtered colimits, we then study sufficient
+conditions for this morphism to be an isomorphism. We say that `C` satisfies the `w`-IPC property if
```
[the remark this answers is not in the document]
Comment (TwoFX) [reply Mathlib/CategoryTheory/Limits/FilteredColimitCommutesProduct.lean:18]: I'm not aware of any, and https://ncatlab.org/nlab/show/IPC-property does not give an alternative name either. I'm definitely not attached to the name though, so I'm happy to take suggestions.

File: Mathlib/CategoryTheory/Limits/FilteredColimitCommutesProduct.lean [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+  simp [← NatTrans.comp_app]
+
+end functorCategory
+
+section
+
+variable (C : Type u) [Category.{v} C]
+
+/-- A category `C` has the `w`-IPC property if the natural morphism
+`colim_k (∏ᶜ s ↦ (F s).obj (k s)) ⟶ ∏ᶜ s ↦ colim_k (F s).obj (k s)` is an isomorpism for any
```
[the remark this answers is not in the document]
Comment (TwoFX) [reply Mathlib/CategoryTheory/Limits/FilteredColimitCommutesProduct.lean]: Fixed, thanks.

**Comment by TwoFX** [comment]
@joelriou I assume you put `WIP` instead of `awaiting-author` by accident, or was that deliberate?

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**